### PR TITLE
fix(mint): validate P2PK/HTLC sigflags and refactor verify_inputs_and_outputs

### PR DESCRIPTION
**Pull request by callebtc**
## Summary

- Reject unsupported `sigflag` values when parsing P2PK and HTLC secrets (`from_secret`), so melts cannot skip meaningful checks via unknown flags.
- Refactor `verify_inputs_and_outputs` so output checks and input/output combined checks run only when `outputs` is provided (not `None`).
- Add mint tests for custom `sigflag` failing `verify_inputs_and_outputs` with melt-style `outputs=None` (with BDHKE/spendable checks patched so spending-condition validation is exercised).

## Files

- `cashu/core/p2pk.py`, `cashu/core/htlc.py` — sigflag validation
- `cashu/mint/verification.py` — verification flow
- `tests/mint/test_mint_conditions.py` — regression tests

### CODE REVIEW

File: cashu/mint/verification.py [hunk 1]
```diff
@@ -83,54 +101,18 @@ async def verify_inputs_and_outputs(
         # Verify SIG_INPUTS spending conditions
         if not all([self._verify_input_spending_conditions(p) for p in proofs]):
             raise TransactionError("validation of input spending conditions failed.")
+        # Verify if proofs are not already spent (this throws)
+        if not await self.db_read._verify_proofs_spendable(proofs):
+            raise InvalidProofsError()
```
Comment (b-l-u-e) [inline cashu/mint/verification.py]: stale  noticed a doc mismatch in verify_inputs_and_outputs,  still says spendability is not checked, but _verify_inputs() now awaits _verify_proofs_spendable(proofs). 
```
 git show HEAD:cashu/mint/verification.py | nl -ba | sed -n '51,57p'
    51          Warning: Does NOT check if the proofs were already spent. Use `db_write._verify_proofs_spendable` for that.
    52
    53          Args:
    54              proofs (List[Proof]): List of proofs to check.
    55              outputs (Optional[List[BlindedMessage]], optional): List of outputs to check.
    56                  Must be provided for a swap but not for a melt. Defaults to None.
    57              conn (Optional[Connection], optional): Database connection. Defaults to None.
```
